### PR TITLE
Create reCaptcha Manager

### PR DESCRIPTION
**Pull request by jbuckner**

### CODE REVIEW

File: test/recaptcha-manager.test.ts [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+      expect(mockGrecaptcha.resetCallCount).to.equal(0);
+      await recaptcha.execute();
+      expect(mockGrecaptcha.resetCallCount).to.equal(1);
+      await recaptcha.execute();
+      expect(mockGrecaptcha.resetCallCount).to.equal(2);
+    });
+
+    // there's no way to know if the user has dismissed the captcha so if another execution
+    // comes through while the captcha is still active, it will be reset
+    it('resets the captcha if another execution gets called', async () => {
```
Comment (iisa) [inline test/recaptcha-manager.test.ts:132]: nice recovery 👍

File: package.json [hunk 2]
```diff
@@ -56,6 +59,9 @@
     "plugins": [
       "@typescript-eslint"
     ],
+    "globals": {
+      "ReCaptchaV2": "readonly"
```
Comment (iisa) [inline package.json:63]: ah, so declaring here allows for [@types/grecaptcha](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/grecaptcha/grecaptcha-tests.ts) to be accessible without import. - took a bit of time to figure that out, was scouring for an `import` 😅

File: src/recaptcha-widget.ts [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+  private setup(
+    container: HTMLElement,
+    widgetConfig?: ReCaptchaV2.Parameters
+  ): void {
+    this.widgetId = this.grecaptchaLibrary.render(container, {
+      callback: this.responseHandler.bind(this),
+      'expired-callback': this.expiredHandler.bind(this),
+      'error-callback': this.errorHandler.bind(this),
+      sitekey: this.siteKey,
+      tabindex: widgetConfig?.tabindex,
```
Comment (iisa) [inline src/recaptcha-widget.ts:105]: q: will the recaptcha render if any of these fields aren't set?

File: src/recaptcha-widget.ts [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+  /** @inheritdoc */
+  async execute(): Promise<string> {
+    const { widgetId } = this;
+
+    if (widgetId === null) {
+      throw new Error('Recaptcha is not setup');
+    }
+
+    if (this.isExecuting) {
+      // there is no way to know when users have dismissed a previous recaptcha
```
Comment (iisa) [inline src/recaptcha-widget.ts:61]: 😢 they don't pass it through via error or expired callbacks?

Comment (iisa) [reply src/recaptcha-widget.ts:61]: q: in user flow, _when_ would this happen? when patron double clicks on button w/ a recaptcha?

**Review by iisa — APPROVED** [review]
cool!

bummer that you can't listen onto a "dismiss" event.  this was like that too in google signup before the api upgrade. way to catch that slip with your recovery reset 🥅 

 test suite is 🔥 too!

File: package.json (hunk 2 above)
Comment (jbuckner) [reply package.json:63]: Yeah this was kind of surprising. I've never had to do this before... you usually just have to install `@types/<package-name>` and it just works, but for some reason with ReCaptcha, I had to add this `globals` section to get it to work.

File: src/recaptcha-widget.ts (hunk 4 above)
Comment (jbuckner) [reply src/recaptcha-widget.ts:61]: Yeah this is frustrating that they don't have a dismiss event! It's such a common activity. This happens if the user opens the captcha then clicks on the backdrop to make it go away. reCaptcha still thinks it's active so you either have to wait for the expired event to fire or just reset it manually the next time it gets called.

File: src/recaptcha-widget.ts (hunk 3 above)
Comment (jbuckner) [reply src/recaptcha-widget.ts:105]: Yeah all of these fields from here down are optional so you only need to override what you want.

**Pull request merged**